adapters/httpapi_curl.c

```diff
@@ -294,9 +309,17 @@ static CURLcode ssl_ctx_callback(CURL *curl, void *ssl_ctx, void *userptr)
         HTTP_HANDLE_DATA *httpHandleData = (HTTP_HANDLE_DATA *)userptr;
 #ifdef USE_OPENSSL
         /*trying to set the x509 per device certificate*/
+        if (httpHandleData->x509privatekeytype == KEY_TYPE_ENGINE) {
+            ENGINE_load_builtin_engines();
+            httpHandleData->engine = ENGINE_by_id(httpHandleData->engineId);
+            if (httpHandleData->engine == NULL)
+            {
+                LogError("unable to load engine by ID: %s", httpHandleData->engineId);
```

Comment:
Basically it wasn't really necessary to set the result, as it would run into the error case when the NULL was passed to x509_openssl_add_credentials below. Nevertheless I changed it to be more clear.

The main intention was, to provide a more detailed error message, otherwise this could have been skipped completely.